File: app/verifiers.py
# -*- coding: utf-8 -*-
# @Time    : 2017/9/5 19:11
# @Author  : Forec
# @File    : verifiers.py
# @Project : WildPointer
# @license : Copyright(C), Forec
# @Contact : forec@bupt.edu.cn

from .models import User
import re


def verify_email(email):
    if len(email) > 7 and re.match("^.+@(\[?)[a-zA-Z0-9\-\.]+\.([a-zA-Z]{2,3}|[0-9]{1,3})(\\]?)$", email) is not None:
        return True
    return False


def verify_username(username):
    if len(username) < 5 or len(username) > 16:
        return False
    invalid = ['?', '/', '=', '>', '<', '!', ',', ';', '.', '\\']
    for inv in invalid:
        if inv in username:
            return False
    return True


def verify_password(password):
    if len(password) < 8 or len(password) > 22:
        return False
    return True


def verify_token(email, token):
    if email is None or token is None or not verify_email(email):
        return None
    user = User.query.filter_by(email=email).first()
    if user is None:
        return None
    if user.verify_password_hash(token):
        return user
    return None

File: app/static/js/utils/verifiers.js

function verifyEmail(email) {
    var emailReg=/^\w+((-\w+)|(\.\w+))*\@[A-Za-z0-9]+((\.|-)[A-Za-z0-9]+)*\.[A-Za-z0-9]+$/i;
    if (email.search(emailReg) == -1)
        return false;
    return true;
}

function verifyPassword(password) {
    if (password == "" || password.length < 8 || password.length > 22)
		return false;
	return true;
}

function verifyUsername(username) {
    var invalid = ['#', '@', '.', ',', '-', '$', '^', '&', '*', '\\', '/', '<', '>', '+', '='];
    for (var i = 0; i < invalid.length; i++) {
        if (username.indexOf(invalid[i]) != -1)
            return false;
    }
	if (username == "" || username.length < 5 || username.length > 32)
		return false;
	return true;
}

function verifyEmailOrUsername(seg) {
    if (seg.indexOf('@') != -1)
        return verifyEmail(seg);
    return verifyUsername(seg);
}